### ISSUE TITLE
Warning false positive with `#[derive(Hash)]` in nightly

### ISSUE THREAD

**Issue by mcarton** [rust-lang/rust#32292]
The following `unused_variables` warning just appeared in Clippy with the last nightly:

```latex
% cat a.rs 
#[derive(Hash)]
struct Foo;

fn main() { let _ = Foo; }
% rustc a.rs 
a.rs:1:10: 1:14 warning: unused variable: `arg_0`, #[warn(unused_variables)] on by default
a.rs:1 #[derive(Hash)]
                ^~~~
a.rs:1:10: 1:14 note: in this expansion of #[derive_Hash] (defined in a.rs)
% rustc --version
rustc 1.9.0-nightly (c66d2380a 2016-03-15)
```
